// Copyright 2019 The MediaPipe Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package mediapipe;

import "mediapipe/framework/calculator.proto";

option objc_class_prefix = "MediaPipe";

message FlowLimiterCalculatorOptions {
  extend mediapipe.CalculatorOptions {
    optional FlowLimiterCalculatorOptions ext = 326963320;
  }

  // The maximum number of frames released for processing at one time.
  // The default value limits to 1 frame processing at a time.
  optional int32 max_in_flight = 1 [default = 1];

  // The maximum number of frames queued waiting for processing.
  // The default value limits to 0 frames awaiting processing.
  optional int32 max_in_queue = 2 [default = 0];

  // The maximum time in microseconds to wait for a frame to finish processing.
  // The default value stops waiting after 1 sec.
  // The value 0 specifies no timeout.
  optional int64 in_flight_timeout = 3 [default = 1000000];
}
